hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/telemetry/TraceExtractingMessageInterceptor.kt

```diff
@@ -23,24 +26,35 @@ import org.springframework.messaging.Message
  */
 class TraceExtractingMessageInterceptor(private val objectMapper: ObjectMapper) : MessageInterceptor<Any> {
   override fun intercept(message: Message<Any>): Message<Any> {
-    val payload = message.payload
-    if (payload !is String) return message
-
-    try {
+    val payload = message.payload as? String
+    // seems to be only true for messages that originated on a topic
+    val span = if (payload?.contains("MessageAttributes") == true) {
       val attributes = objectMapper.readValue(
         objectMapper.readTree(payload).at("/MessageAttributes").traverse(),
         object : TypeReference<MutableMap<String, MessageAttribute>>() {},
       )
-
       val spanName = attributes["eventType"]?.let { "RECEIVE ${it.Value}" } ?: "RECEIVE"
-      val span = attributes.extractTelemetryContext().startSpan(spanName)
+      attributes.extractTelemetryContext().startSpan(spanName)
+    } else {
+      // otherwise we have to grab the attributes from the message
+      // unfortunately these appear to then be not populated for a topic, so have to do both
+      extractAttributes(message)?.let { attributes ->
+        val spanName = attributes["eventType"]?.let { "RECEIVE ${it.stringValue()}" } ?: "RECEIVE"
+        attributes.extractTelemetryContextFromValues().startSpan(spanName)
+      }
+    }
+    return if (span == null) {
+      message
+    } else {
+      message.withHeader("span", span).withHeader("scope", span.makeCurrent())
+    }
+  }
 
-      return message
-        .withHeader("span", span)
-        .withHeader("scope", span.makeCurrent())
-    } catch (e: Exception) {
-      log.error("Not attempting to extract trace context from message: {} with headers: {} due to exception {}", message.payload, message.headers, e.message)
-      return message
```

Comment:
Are you sure about dropping the try/catch? I think it means if we get an unexpected exception we'll drop the message entirely and won't realise unless someone notices it didn't arrive.

Reply:
Oh they might end up on the DLQ? 🤔

Reply:
added back in the try catch